src/WebJobs.Script/Description/DocumentDBBindingMetadata.cs

```diff
@@ -25,9 +22,9 @@ public override void ApplyToConfig(JobHostConfigurationBuilder configBuilder)
             }
 
             DocumentDBConfiguration config = new DocumentDBConfiguration();
-            if (!string.IsNullOrEmpty(ConnectionString))
+            if (!string.IsNullOrEmpty(Connection))
             {
-                config.ConnectionString = ConnectionString;
+                config.ConnectionString = Utility.GetAppSettingOrEnvironmentValue(Connection);
             }
 
```

Comment:
It gets called for each binding: https://github.com/Azure/azure-webjobs-sdk-script/blob/master/src/WebJobs.Script/Host/ScriptHost.cs#L216